components/opponent/commons/opponent_display.lua

```diff
@@ -106,10 +121,18 @@ OpponentDisplay.propTypes.InlineOpponent = {
 	teamStyle = TypeUtil.optional(OpponentDisplay.types.TeamStyle), -- only affects Opponent.team
 }
 
---[[
-Displays an opponent as an inline element. Useful for describing opponents in
-prose.
-]]
+---@class InlineOpponentProps
+---@field flip boolean?
+---@field opponent standardOpponent
+---@field showFlag boolean?
+---@field showLink boolean?
+---@field dq boolean?
+---@field teamStyle teamStyle
+
+---Displays an opponent as an inline element. Useful for describing opponents in prose.
+---comment
+---@param props InlineOpponentProps
+---@return Html|string|nil
```

Comment:
```suggestion
---@return Html|string
```
How does it return nil?

Reply:
because `OpponentDisplay.InlineTeamContainer` could return nothing